snuba/snuba_migrations/spans/0001_spans_v1.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, DateTime, Nested, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.schemas import Float
+
+storage_set_name = StorageSetKey.SPANS
+local_table_name = "spans_local"
+dist_table_name = "spans_dist"
+
+UNKNOWN_SPAN_STATUS = 2
+
+columns: List[Column[Modifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("transaction_id", UUID(Modifiers(nullable=True))),
+    Column("trace_id", UUID()),
+    Column("span_id", UInt(64)),
+    Column("parent_span_id", UInt(64, Modifiers(nullable=True))),
+    Column("segment_id", UInt(64)),
+    Column("is_segment", UInt(8)),
+    Column("segment_name", String(Modifiers(default="''"))),
+    Column("start_timestamp", DateTime()),
+    Column("end_timestamp", DateTime()),
+    Column("timestamp", DateTime(Modifiers(materialized="end_timestamp"))),
+    Column("duration", UInt(32)),
+    Column("exclusive_time", Float(64)),
+    Column("op", String(Modifiers(low_cardinality=True))),
+    Column("group", UInt(64)),
+    Column("span_status", UInt(8)),
+    Column("span_kind", String(Modifiers(low_cardinality=True))),
+    Column("description", String()),
+    Column("status", UInt(32, Modifiers(nullable=True))),
+    Column("module", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("action", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("domain", String(Modifiers(nullable=True))),
+    Column("platform", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("user", String(Modifiers(nullable=True))),
+    Column("tags", Nested([("key", String()), ("value", String())])),
+    Column(
+        "measurements",
+        Nested(
+            [("key", String(Modifiers(low_cardinality=True))), ("value", Float(64))]
+        ),
+    ),
+    Column("partition", UInt(16)),
+    Column("offset", UInt(64)),
+    Column("retention_days", UInt(16)),
+    Column("deleted", UInt(8)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set_name,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    order_by="(project_id, segment_name, segment_id, timestamp)",
```

Comment:
> From https://github.com/getsentry/snuba/pull/4081#discussion_r1178527181, Filippo: If segment_id comes before timestamp, the timestamp field is not going to be used in a query across different segments. I think you will not want to do that.
Do we have a segment start timestamp ? to do something like: segment_start_ts, segment_id, teimstamp ?
Or something like toStartOfHour(timestamp), segment_id, timestamp

Not all spans would have a corresponding segment start timestamp. Only the `segment_name` and `segment_id` are copied to all child spans in the existing model. 

The current schema is optimized for searching for outlier `segment_id` of a `segment` based on different criteria's. Duration, module etc. @wmak, @shruthilayaj, @gggritso is this a fair assumption to make? Are your workflows around sampled spans going to be around this idea? If not, what would the workflow be?

Reply:
@nikhars the current workflow for the modules is currently narrowing down to a specific span description(eg. a SQL query) using things like throughput, module, domain and action.
Then using that description to find corresponding transaction names
Then showing example span events (rows in this table) based on everything above

Reply:
Normalized, the workflows actually depend on the description being normalized because otherwise we can't show useful groups of things. I think at most (we dont have anything for this yet) we just want to display the raw description next to the normalized one, but we can use eventstore for that

Reply:
Just reiterating to be on the same page. Please confirm
From Metrics - the current workflow for the modules is currently narrowing down to a specific span description(eg. a SQL query) using things like throughput, module, domain and action.
From Metrics - Then using that description to find corresponding transaction names
From Spans table - Then showing example span events (rows in this table) based on everything above

Now, for showing example span events, the workflows imply that most queries would have a where condition like this, correct?
```
WHERE project_id = A AND segment_name = B
```